app/javascript/components/forms/SharedAccessForm.jsx

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable react/jsx-props-no-spreading */
+
+import React, { useState } from 'react';
+import {
+  Button, Col, Form, Row, Stack,
+} from 'react-bootstrap';
+import { useForm } from 'react-hook-form';
+import PropTypes from 'prop-types';
+import { useParams } from 'react-router-dom';
+import useShareAccess from '../../hooks/mutations/shared_accesses/useShareAccess';
+import Avatar from '../users/Avatar';
+import SearchBar from '../shared/SearchBar';
+import useShareableUsers from '../../hooks/queries/shared_accesses/useShareableUsers';
+
+export default function SharedAccessForm({ handleClose }) {
+  const { register, handleSubmit } = useForm();
+  const { friendlyId } = useParams();
+  const { onSubmit } = useShareAccess({ friendlyId, closeModal: handleClose });
+  const { data: users } = useShareableUsers(friendlyId);
+  const [search, setSearch] = useState('');
+
+  return (
+    <>
+      <SearchBar id="shared-users-modal-search" setSearch={setSearch} />
+      <Form onSubmit={handleSubmit(onSubmit)}>
+        <Row className="border-bottom pt-3 pb-2">
+          <Col>
+            <span className="text-muted small"> Name </span>
+          </Col>
+          <Col>
+            <span className="text-muted small"> Email address </span>
+          </Col>
+        </Row>
+        {
+          users?.filter((user) => {
+            if (user.name.toLowerCase().includes(search.toLowerCase())) {
+              return user;
+            }
+            return false;
+          }).map((user) => (
+            <Row className="border-bottom py-3" key={user.id}>
+              <Col>
+                <Stack direction="horizontal">
+                  <Form.Check
+                    type="checkbox"
+                    value={user.id}
+                    aria-label="tbd"
+                    className="pe-3"
+                    {...register('users')}
+                  />
+                  <Avatar avatar={user.avatar} radius={40} />
+                  <h6 className="text-primary mb-0 ps-3"> { user.name } </h6>
+                </Stack>
+              </Col>
+              <Col className="my-auto">
+                <span className="text-muted"> { user.email } </span>
+              </Col>
+            </Row>
+          ))
+        }
+        <Stack className="mt-3" direction="horizontal" gap={1}>
+          <Button variant="primary-light" className="ms-auto" onClick={handleClose}>
+            Close
+          </Button>
+          <Button variant="primary" type="submit">
+            Share
+          </Button>
+        </Stack>
+      </Form>
+    </>
+  );
+}
+
+SharedAccessForm.propTypes = {
+  handleClose: PropTypes.func,
```

Comment:
Shouldn't this be `required`? Then you can remove the defaultProps

Reply:
It does this warning in console if handleClose is set as required:
`Warning: Failed prop type: The prop `handleClose` is marked as required in `SharedAccessForm`, but its value is `undefined`.`

Reply:
We can't set it as isRequired because of the way the modal template is made.

It will throw a constant Warning in the console:

> Warning: Failed prop type: The prop `handleClose` is marked as required in `DeleteUserForm`, but its value is `undefined`.

It was acknowledged in #3405 you can see Amir's explanation below.

> I think this should change to handleColse: PropTypes.func because you are creating the DeleteUserForm first without the handleClose prop and inject afterwards with the cloneElement API.